awx/main/wsrelay.py

```diff
@@ -323,7 +336,7 @@ async def run(self):
                 logger.info(f"Adding {new_remote_hosts} to websocket broadcast list")
 
             for h in deleted_remote_hosts:
-                self.cleanup_offline_host(h)
+                await self.cleanup_offline_host(h)
```

Comment:
Considering that timeouts are a possibility, you might as well run this simultaneously for all hosts in `deleted_remote_hosts`